File: output/ChatGPT/codenet/Java/C++/s853705214.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> nums(n);

    for (int i = 0; i < n; i++) {
        cin >> nums[i];
    }

    int min = INT_MAX;
    int countNegatives = 0;
    long long sum = 0;
    for (int num : nums) {
        int abs = abs(num);
        sum += abs;
        min = std::min(min, abs);
        if (num < 0) {
            countNegatives++;
        }
    }

    cout << (countNegatives % 2 == 0 ? sum : sum - (min*2)) << endl;

    return 0;
}

File: output/ChatGPT/codenet/Java/Python/s853705214.py
import sys

n = int(input())
nums = list(map(int, sys.stdin.readline().split()))

min_val = float('inf')
count_negatives = 0
sum_val = 0
for num in nums:
    abs_val = abs(num)
    sum_val += abs_val
    min_val = min(min_val, abs_val)
    if num < 0:
        count_negatives += 1

print(sum_val if count_negatives % 2 == 0 else sum_val - (min_val*2))